pkg/deploy/api/validation/validation.go

```diff
@@ -123,33 +127,56 @@ func validateCustomParams(params *deployapi.CustomDeploymentStrategyParams, fldP
 	return errs
 }
 
-func validateRecreateParams(params *deployapi.RecreateDeploymentStrategyParams, fldPath *field.Path) field.ErrorList {
+func validateRecreateParams(params *deployapi.RecreateDeploymentStrategyParams, pod *kapi.PodSpec, fldPath *field.Path) field.ErrorList {
 	errs := field.ErrorList{}
 
+	if params.TimeoutSeconds != nil && *params.TimeoutSeconds < 1 {
+		errs = append(errs, field.Invalid(fldPath.Child("timeoutSeconds"), *params.TimeoutSeconds, "must be >0"))
+	}
+
 	if params.Pre != nil {
-		errs = append(errs, validateLifecycleHook(params.Pre, fldPath.Child("pre"))...)
+		errs = append(errs, validateLifecycleHook(params.Pre, pod, fldPath.Child("pre"))...)
 	}
 	if params.Mid != nil {
-		errs = append(errs, validateLifecycleHook(params.Mid, fldPath.Child("mid"))...)
+		errs = append(errs, validateLifecycleHook(params.Mid, pod, fldPath.Child("mid"))...)
 	}
 	if params.Post != nil {
-		errs = append(errs, validateLifecycleHook(params.Post, fldPath.Child("post"))...)
+		errs = append(errs, validateLifecycleHook(params.Post, pod, fldPath.Child("post"))...)
 	}
 
 	return errs
 }
 
-func validateLifecycleHook(hook *deployapi.LifecycleHook, fldPath *field.Path) field.ErrorList {
+func validateLifecycleHook(hook *deployapi.LifecycleHook, pod *kapi.PodSpec, fldPath *field.Path) field.ErrorList {
 	errs := field.ErrorList{}
 
 	if len(hook.FailurePolicy) == 0 {
 		errs = append(errs, field.Required(fldPath.Child("failurePolicy"), ""))
 	}
 
-	if hook.ExecNewPod == nil {
-		errs = append(errs, field.Required(fldPath.Child("execNewPod"), ""))
-	} else {
+	switch {
+	case hook.ExecNewPod != nil && len(hook.TagImages) > 0:
+		errs = append(errs, field.Invalid(fldPath, "<hook>", "only one of 'execNewPod' of 'tagImages' may be specified"))
+	case hook.ExecNewPod != nil:
 		errs = append(errs, validateExecNewPod(hook.ExecNewPod, fldPath.Child("execNewPod"))...)
+	case len(hook.TagImages) > 0:
+		for i, image := range hook.TagImages {
+			if len(image.ContainerName) == 0 {
+				errs = append(errs, field.Required(fldPath.Child("tagImages").Index(i).Child("containerName"), "a containerName is required"))
+			} else {
+				if _, err := deployapi.TemplateImageForContainer(pod, deployapi.IgnoreTriggers, image.ContainerName); err != nil {
+					errs = append(errs, field.Invalid(fldPath.Child("tagImages").Index(i).Child("containerName"), image.ContainerName, err.Error()))
```

Comment:
We don't do that in all places, and names can be very long, so I'm not sure
we need to start introducing it here.

On Thu, Feb 11, 2016 at 8:30 AM, David Eads <notifications@github.com>
wrote:

> In pkg/deploy/api/validation/validation.go
> <https://github.com/openshift/origin/pull/6925#discussion_r52600613>:
>
> >  		errs = append(errs, validateExecNewPod(hook.ExecNewPod, fldPath.Child("execNewPod"))...)
> > +	case len(hook.TagImages) > 0:
> > +		for i, image := range hook.TagImages {
> > +			if len(image.ContainerName) == 0 {
> > +				errs = append(errs, field.Required(fldPath.Child("tagImages").Index(i).Child("containerName"), "a containerName is required"))
> > +			} else {
> > +				if _, err := deployapi.TemplateImageForContainer(pod, deployapi.IgnoreTriggers, image.ContainerName); err != nil {
> > +					errs = append(errs, field.Invalid(fldPath.Child("tagImages").Index(i).Child("containerName"), image.ContainerName, err.Error()))
>
> You should be able to give me back a list of valid values. NotSupported
> is the wrong connotation, but the right shape of the error.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/6925/files#r52600613>.
>